README.md

```diff
@@ -19,10 +19,86 @@ build and deployment processes.
 
 [**Documentation is Available Online!**][3]
 
+# Connection Pool
+
+The NuoDB Node.js driver comes with a built in connection pool available.
+
+### Arguments
+
+**connectionConfig:** the configuration that will be used to create the connection in the pool, required argument.
+
+**minAvailable:** initial size of the pool and the number of connection the pool will aim to maintain, default of 10 is used if no argument is provided.
+
+**maxAge:** Amount of time from connection creation until it will age out, default of 300000ms (5 minutes) is used if no argument is provided.
+
+**checkTime:** how often the pool will run an internal liveliness check on free connections, default of 120000ms(2 minutes) is used if no argument is provided.
+
+**maxLimit:** hard cap on the amount of live connection the pool can maintain, default of 200 is used if no argument is provided. If 0, the pool will have no hard cap.
+
+**connectionRetryLimit:** amount of times a pool will attempt to create a connection, default of 5 is used if no argument is provided.
```

Comment:
Is there any spacing between the retries?  If not we should probably add something to control that, even add the possibility the time between tries increases.  If we can't get a connection it probably means something needs to be started or something needs time to finish, so spacing the retries so they are not rapid in succession would probably be helpful.